core/lib/dal/src/eth_sender_dal.rs

```diff
@@ -18,122 +19,108 @@ pub struct EthSenderDal<'a, 'c> {
+             ORDER BY id"
+        )
+        .fetch_all(self.storage.conn())
+        .await?;
+        Ok(txs.into_iter().map(|tx| tx.into()).collect())
     }
 
-    pub async fn get_eth_l1_batches(&mut self) -> L1BatchEthSenderStats {
-        {
-            let mut stats = L1BatchEthSenderStats::default();
-            for tx_type in ["execute_tx", "commit_tx", "prove_tx"] {
-                let mut records= sqlx::query(&format!(
-                        "SELECT number as number, true as confirmed FROM l1_batches
-                         INNER JOIN eth_txs_history ON (l1_batches.eth_{}_id = eth_txs_history.eth_tx_id)
-                         WHERE eth_txs_history.confirmed_at IS NOT NULL
-                         ORDER BY number DESC
-                         LIMIT 1",
-                        tx_type
-                    ))
-                        .fetch_all(self.storage.conn())
-                        .await
-                        .unwrap();
-
-                records.extend(sqlx::query(&format!(
-                    "SELECT number as number, false as confirmed FROM l1_batches
+    pub async fn get_eth_l1_batches(&mut self) -> sqlx::Result<L1BatchEthSenderStats> {
+        let mut stats = L1BatchEthSenderStats::default();
+        for tx_type in ["execute_tx", "commit_tx", "prove_tx"] {
+            let mut records= sqlx::query(&format!(
+                    "SELECT number as number, true as confirmed FROM l1_batches
                      INNER JOIN eth_txs_history ON (l1_batches.eth_{}_id = eth_txs_history.eth_tx_id)
+                     WHERE eth_txs_history.confirmed_at IS NOT NULL
                      ORDER BY number DESC
                      LIMIT 1",
                     tx_type
                 ))
                     .fetch_all(self.storage.conn())
-                    .await
-                    .unwrap());
-
-                for record in records {
-                    let batch_number = L1BatchNumber(record.get::<i64, &str>("number") as u32);
-                    let aggregation_action = match tx_type {
-                        "execute_tx" => AggregatedActionType::Execute,
-                        "commit_tx" => AggregatedActionType::Commit,
-                        "prove_tx" => AggregatedActionType::PublishProofOnchain,
-                        _ => unreachable!(),
-                    };
-                    if record.get::<bool, &str>("confirmed") {
-                        stats.mined.push((aggregation_action, batch_number));
-                    } else {
-                        stats.saved.push((aggregation_action, batch_number));
-                    }
+                    .await?;
+
+            records.extend(
+                sqlx::query(&format!(
+                    "SELECT number as number, false as confirmed FROM l1_batches
+                 INNER JOIN eth_txs_history ON (l1_batches.eth_{}_id = eth_txs_history.eth_tx_id)
+                 ORDER BY number DESC
+                 LIMIT 1",
```

Comment:
Looks like the formatting has been broken here.

Reply:
done